### PR TITLE
changing thunk to promise seems to gain better performance 

### PR DESCRIPTION
**Pull request by renaesop**
1. change thunk to promise;
2. make the disable cache option can be applied to partials

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -41,9 +41,12 @@ var rLayoutPattern = /{{!<\s+([A-Za-z0-9\._\-\/]+)\s*}}/;
  */
 
 var read = function (filename) {
-  return function(done) {
-    fs.readFile(filename, {encoding: 'utf8'}, done);
-  };
+  return new Promise(function(resolve, reject){
+    fs.readFile(filename, {encoding: 'utf8'}, function (err, rs) {
```
Comment (coderhaoxin) [inline index.js]: we can use `mz`, https://github.com/normalize/mz

Comment (renaesop) [inline index.js]: yeah! It looks good!

**Comment by shellscape** [comment]
@renaesop looks like the tests need to be updated for travis to pass.

also, this will be an unpopular position, given that the new hotness is to move away from thunks while we wait for async/await to be finalized and implemented: there are lots of people still using thunks. the changes are small enough that we can and should provide and option for which return pattern the user would like.

**Comment by shellscape** [comment]
@renaesop any chance you could pull out your changes and the caching fixes into a separate PR? If not, I can cherry pick them. but I'd like to get those fixes in while we wait for feedback on switching to promises from thunks.

**Comment by shellscape** [comment]
cherry picked the partials cache fix, but at this time are not proceeding with the promise implementation. at a point in the future we'll go that route for koa@2 compatibility.

**Pull request closed without merging**